Libraries/Share/Share.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule Share
+ * @flow
+ */
+'use strict';
+
+const Platform = require('Platform');
+const {
+  ActionSheetManager,
+  ShareModule
+} = require('NativeModules');
+const invariant = require('fbjs/lib/invariant');
+const processColor = require('processColor');
+
+type Content = { title?: string, message: string } | { title?: string, url: string };
+type Options = { dialogTitle?: string, excludeActivityTypes?: Array<string>, tintColor?: string };
+
+class Share {
+
+  /**
+   * Open a dialog to share text content.
+   * 
+   * In iOS, Returns a Promise which will be invoked an object containing `action`, `activityType`.
+   * If the user dismissed the dialog, the Promise will still be resolved with action being `Share.dismissedAction` 
```

Comment:
How about a `isCancelled` property instead? cc @grabbou we discussed about this.